### PR TITLE
Create lease

### PR DESCRIPTION
**Pull request by AmudaPalani**
## Proposed changes

Refactor create lease controller to use new data model

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [x] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: pkg/lease/service.go [hunk 1]
```diff
@@ -122,6 +131,84 @@ func (a *Service) List(query *Lease) (*Leases, error) {
...
+
+	existingLeases, err := a.List(query)
+	if err != nil {
+		return nil, errors.NewInternalServer("lease", err)
+	}
+	if existingLeases != nil && len(*existingLeases) > 0 {
+		return nil, errors.NewAlreadyExists("lease", *data.PrincipalID)
+	}
+
+	new := NewLease(NewLeaseInput{
```
[the remark this answers is not in the document]
Comment (marinatedpork) [reply pkg/lease/service.go]: This logic of going from `Lease` to `NewLeaseInput` to `Lease` just to stick the ID and status in there feels kind of funny, but not a blocker or anything.

File: pkg/lease/model.go [hunk 2]
```diff
@@ -114,3 +116,36 @@ func (c StatusReason) StatusReasonPtr() *StatusReason {
 	v := c
 	return &v
 }
+
+// StringPtr returns a pointer to the input string value
+func StringPtr(c string) *string {
```
Comment (marinatedpork) [inline pkg/lease/model.go]: Also, this generic method for take a string and returning a pointer might be better off in a generic package as opposed to the model package.

File: cmd/lambda/leases/create_test.go [hunk 3]
```diff
@@ -2,673 +2,373 @@ package main
...
+		request      events.APIGatewayProxyRequest
+		retLease     *lease.Lease
+		retAccounts  *account.Accounts
+		retAccount   *account.Account
+		retListErr   error
+		retUpdateErr error
+		retCreateErr error
+	}{
+		{
+			name: "When principalId is missing. Then an internal server error is returned.",
```
[the remark this answers is not in the document]
Comment (AmudaPalani) [reply cmd/lambda/leases/create_test.go]: fixed

File: cmd/lambda/leases/create.go [hunk 4]
```diff
@@ -2,203 +2,101 @@ package main
...
+		spent = spent + *usageItem.CostAmount
 	}
-	leaseMsg := string(leaseMsgBytes)
 
-	// Publish message to the lease topic on the success of the lease creation
-	log.Printf("Sending Lease Message to SNS Topic %s\n", *topic)
-	messageID, err := snsSvc.PublishMessage(topic, &leaseMsg, true)
+	// Create lease
+	newLease.AccountID = leasedAccount.ID
+	lease, err := Services.LeaseService().Create(newLease, spent)
```
[the remark this answers is not in the document]
Comment (AmudaPalani) [reply cmd/lambda/leases/create.go]: fixed

File: pkg/lease/service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AmudaPalani) [reply pkg/lease/service.go]: fixed variable name

File: pkg/lease/model.go (hunk 2 above)
Comment (AmudaPalani) [reply pkg/lease/model.go]: This is unused function, so I removed it.

**Pull request closed without merging**